src/bootstrap/test.rs

```diff
@@ -1802,6 +1803,10 @@ impl Step for CrateRustdoc {
         cargo.arg("--");
         cargo.args(&builder.config.cmd.test_args());
 
+        if target.contains("musl") {
+            cargo.arg("'-Ctarget-feature=-crt-static'");
```

Comment:
The extra `'` quotes here look somewhat suspicious, is this being exercised somewhere?

Reply:
IIRC when the argument was passed to some command (rustdoc?) it was called like `"command" "-Ctarget-feature=-crt-static"` and it was complaining about unknown option option or something alike. The error magically disappeared when I called it with additional `'` quotes.
I'll run local build without it to be sure and post the error if any.

Reply:
Okay here is the error without `'`:
```
   Compiling rustdoc v0.0.0 (/checkout/src/librustdoc)
[RUSTC-TIMING] rustdoc test:true 77.706
[RUSTC-TIMING] rustdoc test:false 151.906
    Finished release [optimized] target(s) in 2m 42s
     Running build/x86_64-unknown-linux-musl/stage2-tools/x86_64-unknown-linux-musl/release/deps/rustdoc-c734edc0b28f9562
error: Unrecognized option: 'C'
error: test failed, to rerun pass '--lib'


command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "test" "--target" "x86_64-unknown-linux-musl" "-j" "16" "--release" "--locked" "--manifest-path" "/checkout/src/tools/rustdoc/Cargo.toml" "-p" "rustdoc:0.0.0" "--" "-Ctarget-feature=-crt-static"
expected success, got: exit code: 101
```

Reply:
Hm this is pretty worrisome and seems indicative of a bug somewhere else along the way...

I'm not entirely comfortable with this but I don't really know of an easy fix for this, so can a comment be added indicating that the single quotes are at least intentional but it's unknown why they're here?

Reply:
It can be even reverted if we want to test `musl` only as a static target for now.
I'll add a comment when I'll be preparing config option for using `musl` as a host for test.